pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SchemaUtils.java

```diff
@@ -147,6 +147,9 @@ public static void validate(Schema schema, boolean isIgnoreCase) {
             .equals(FieldSpec.DataType.DOUBLE)) {
           validateDefaultIsNotNaN(fieldSpec);
         }
+        if (!fieldSpec.isSingleValueField()) {
```

Comment:
Not introduced in this PR, but we should probably not skip the check for virtual columns

Reply:
Yeah. I'm thinking just remove that if check

Reply:
Virtual column means the data are not ingested physically. It should follow all restrictions

Reply:
followup here: https://github.com/apache/pinot/pull/16091